html.md

```diff
@@ -1,68 +1,83 @@
 # HTML
 
- 1. **[Attribute Ordering](#attribute-ordering-within-tags)**
+## Table of Contents
+ 1. **[Attribute Ordering](#attribute-ordering-within-tags)** [ [Attribute Order](#11-attribute-order) ]
  1. **[Quotes](#quotes)**
- 1. **[Buttons](#buttons)**
+ 1. **[Inputs](#inputs)**
+ 1. **[Buttons](#buttons)** [ [Submit Buttons](#31-submit-buttons) ] [ [Links](#32-links) ] [ [Buttons](#33-buttons) ]
  1. **[Forms](#forms)**
- 1. **[Comments](#comments)**
- 1. **[Semantics](#semantics)**
+ 1. **[Comments](#comments)** [ [When Things Get Nesty](#51-when-things-get-nesty) ]
+ 1. **[Semantics](#semantics)** [ [`<strong>` vs `<b>`](#61-strong-vs-b) ] [ [`<em>` vs `<i>`](#62-em-vs-i) ]
  1. **[Do NOT Use](#do-not-use)**
 
-<a name="attribute-ordering-within-tags"></a>
-## 1. Attribute Ordering within Tags
+
+## Attribute Ordering
  > Makes code easier to read and easier for a new developer to jump in and find what they are looking for.
 
-  ### 1.1 Attribute Order
-   Ordering from the beginning of the tag to the close of the tag is as follows:
+### 1.1 Attribute Order
+Ordering from the beginning of the tag to the close of the tag is as follows:
 
-  1. id/for
-  1. className
-  1. tag specific (eg: href, type, action)
-  1. element state (eg: value, checked, readonly)
-  1. ARIA/accessibility
-  1. data- attributes
+1. id/for
+1. className
+1. tag specific (eg: href, type, action)
+1. element state (eg: value, checked, readonly)
+1. ARIA/accessibility
+1. data- attributes
 
 
-**Bad**
 ```html
-    <input type='...' class='...' id='...' .... etc. />
+<!-- Bad -->
+<input type='...' class='...' id='...' .... etc. />
 ```
 
-**Good**
 ```html
-    <input id='...' class='...' type='...' ... etc. />
+<!-- Good -->
+<input id='...' class='...' type='...' ... etc. />
 ```
 
-<a name="quotes"></a>
-## 2. Quotes
-  > Hint - it's easier to type single quotes.
-
-+ Use whatever is comfortable to you
-+ Do NOT go back to change code to suit your style
+## Quotes
+> Hint - it's easier to type single quotes.
+- Use whatever is comfortable to you
+- Do NOT go back to change code to suit your style
 
-**Bad**
 ```html
-<div class="class1"> //changing this to single quotes to fit your style
+<!-- Bad -->
+<div class="class1"> <!-- do not change -->
     <p class='class2'>This is paragraph content.</p>
 </div>
 ```
 
-**Good**
 ```html
+<!-- Good -->
 <div class='class1'>
     <p class='class2'>This is paragraph content.</p>
 </div>
 
-//or
-
 <div class="class1">
     <p class="class2">This is paragraph content.</p>
 </div>
 ```
 
-<a name="buttons"></a>
-## 3. Buttons
-  > Buttons can submit a form, be a link styled like a button, or simply just a button.
+## Inputs
+All inputs and labels must have an explicit relationship
```

Comment:
I'm going to be that guy - missing a period at the end :-)

Reply:
.